File: spotifyarch/src/main/java/com/timothy/spotifyarchitecture/retrofit/models/CursorPager.java
package com.timothy.spotifyarchitecture.retrofit.models;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

/**
 * <a href="https://developer.spotify.com/web-api/object-model/#cursor-based-paging-object">Cursor-based paging object model</a>
 *
 * @param <T> expected object that is paged
 */
public class CursorPager<T> {
    public String href;
    public List<T> items;
    public int limit;
    public String next;
    public Cursor cursors;
    public int total;

    public CursorPager() {
    }
}

File: spotifyarch/src/main/java/com/timothy/spotifyarchitecture/retrofit/models/CursorPager.kt
package com.timothy.spotifyarchitecture.retrofit.models

/**
 * [Cursor-based paging object model](https://developer.spotify.com/web-api/object-model/#cursor-based-paging-object)

 * @param <T> expected object that is paged
</T> */
class CursorPager<T> {
    var href: String = ""
    var items: List<T> = listOf()
    var limit: Int = 0
	var next: String? = ""
    var cursors: Cursor? = null
    var total: Int = 0
}
